app/(api)/_datalib/collections/updateitem.ts

```diff
@@ -0,0 +1,40 @@
+import { getClient } from '@utils/mongodb/mongoClient.mjs';
+import { NotFoundError } from '../../_utils/response/Errors';
+import { NoContentError } from '@utils/response/Errors';
+import isBodyEmpty from '@utils/request/isBodyEmpty';
+
+export async function updateCollectionItem(
+  collectionName: never,
+  id: string,
+  body: object
+) {
+  try {
+    if (!id || isBodyEmpty(body)) {
+      throw new NoContentError();
+    }
+
+    const client = await getClient();
+    const db = client.db;
+    const collection = db.collection(collectionName);
+
+    const result = await collection.findOneAndUpdate({ _id: id }, body);
```

Comment:
Small error here. `id` comes in as a string, but MongoDb (being annoying) does not match ids using strings. They need you to convert the id to an ObjectId first :(. You can look at some examples of us doing this either in this codebase or in:

https://github.com/HackDavis/judging-app-2024/tree/main/app/(api)

Additionally, I wrote this function called parseAndReplace. You can read up on what it does at the bottom of the readme, but basically you should always just run body through parseAndReplace at some point. Even if the functionality isn't used, you can just pass it through the function since it doesn't change regular json objects.